pkg/il/evaluator/evaluator.go

```diff
@@ -183,19 +186,29 @@ func (e *IL) getAttrContext() *attrContext {
 }
 
 func (e *IL) evalResult(expr string, attrs attribute.Bag) (interpreter.Result, error) {
+	ctx := e.getAttrContext()
+	return ctx.evalResult(expr, attrs, e.maxStringTableSizeForPurge)
+}
+
+func (ctx *attrContext) evalResult(
+	expr string, attrs attribute.Bag, maxStringTableSizeForPurge int) (interpreter.Result, error) {
+
 	var entry cacheEntry
 	var err error
-	if entry, err = e.getOrCreateCacheEntry(expr); err != nil {
+	if entry, err = ctx.getOrCreateCacheEntry(expr); err != nil {
 		glog.Infof("evaluator.evalResult failed expr:'%s', err: %v", expr, err)
 		return interpreter.Result{}, err
 	}
 
-	return entry.interpreter.Eval("eval", attrs)
+	r, err := entry.interpreter.Eval("eval", attrs)
+	if entry.interpreter.StringTableSize() > maxStringTableSizeForPurge {
+		ctx.cache.Remove(expr)
```

Comment:
If I understand correctly, this means that once the max size is hit, no new expressions will be added and so new expressions run "slower" for ever more.

Would it be possible to purge an entry from the cache at random instead? This doesn't require any LRU-tracking logic, but would generally behave better in the long run.